back/app/models/user.rb

```diff
@@ -167,17 +168,24 @@ def self.from_token_request(request)
     email = request.params['auth']['email']
 
     # Hack to embed phone numbers in email
-    if AppConfiguration.instance.feature_activated?('password_login') && AppConfiguration.instance.settings(
-      'password_login', 'phone'
-    )
-      phone_service = PhoneService.new
-      if phone_service.phone_or_email(email) == :phone
-        pattern = AppConfiguration.instance.settings('password_login', 'phone_email_pattern')
-        email = pattern.gsub('__PHONE__', phone_service.normalize_phone(email))
-      end
-    end
+    phone_or_email = PhoneService.new.emailize_email_or_phone(email)
+
+    not_invited.find_by_cimail(phone_or_email)
+  end
+
+  def assign_email_or_phone
+    # Hack to embed phone numbers in email
+    phone_or_email = PhoneService.new.emailize_email_or_phone(email)
```

Comment:
I'm also a bit concerned if this will keep working well when e.g. phone numbers are included in tenant templates this will break those templates as the phone number might already be in the correct format but then would be wrapped inside this format a second time? But probably we're already having issues with this today as the app config is currently separated from the templates.